astropy/io/ascii/cds.py

```diff
@@ -164,14 +205,486 @@ def get_cols(self, lines):
                     raise ValueError(f'Line "{line}" not parsable as CDS header')
 
         self.names = [x.name for x in cols]
-
         self.cols = cols
 
+    def _split_float_format(self, value):
+        """
+        Splits a Float string into different parts to find number
+        of digits after decimal and check if the value is in Scientific
+        notation.
+
+        Parameters
+        ----------
+        value : the float value
+
+        Returns
+        -------
+        fmt: (int, int, int, bool, bool)
+            List of values describing the Float sting.
+            (size, dec, ent, sign, exp)
+            size, length of the given string.
+            ent, number of digits before decimal point.
+            dec, number of digits after decimal point.
+            sign, whether or not given value signed.
+            exp, is value in Scientific notation?
+        """
+        regfloat = re.compile(r"""(?P<sign> [+-]*)
+                                  (?P<ent> [^eE.]+)
+                                  (?P<deciPt> [.]*)
+                                  (?P<decimals> [0-9]*)
+                                  (?P<exp> [eE]*-*)[0-9]*""",
+                              re.VERBOSE)
+        mo = regfloat.match(value)
+
+        if mo is None:
+            raise Exception(f'{value} is not a float number')
+        return (len(value),
+                len(mo.group('ent')),
+                len(mo.group('decimals')),
+                mo.group('sign') != "",
+                mo.group('exp') != "")
+
+    def _set_column_val_limits(self, col):
+        """
+        Sets the `col.min` and `col.max` column attributes,
+        taking into account columns with Null values.
+        """
+        col.max = max(col)
+        col.min = min(col)
+        if col.max is np.ma.core.MaskedConstant:
+            col.max = None
+        if col.min is np.ma.core.MaskedConstant:
+            col.min = None
+
+    def column_float_formatter(self, col):
+        """
+        String formatter for a column containing Float values.
+        """
+        # maxsize: maximum length of string containing the float value.
+        # maxent: maximum number of digits places before decimal point.
+        # maxdec: maximum number of digits places after decimal point.
+        # maxprec: maximum precision of the column values, sum of maxent and maxdec.
+        maxsize, maxprec, maxent, maxdec = 1, 0, 1, 0
+        sign = False
+        fformat = 'F'
+
+        # Find maximum sized value in the col
+        if getattr(col.meta, 'size', None) is not None:  # If ``formats`` passed.
```

Comment:
I know that the `formats` argument is being used to mention the right precision for values. But, does finding the `size` attribute in `col.meta` automatically imply that it is for ``formats``? Should there be an additional check? If not, you could make it clear in the docstring/docs that this is expected.

Reply:
It is only when the `formats` argument is passed that the `col.meta.size` is set prior to checking and formatting for different datatypes. See [L407-408](https://github.com/astropy/astropy/pull/11897/files#diff-f1aebe08fb4f3dd370b7dec776d172228663c8a0ae39b8115e9f678b8199e42bR407-R408).
I will make it more clear in the docstring.